### PR TITLE
feat: update go-ipld-format and boxo to remove globals

### PR DESCRIPTION
**Pull request by aschmahmann**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -6,11 +6,11 @@ require (
 	github.com/dgraph-io/badger v1.6.2
 	github.com/golang/protobuf v1.5.2
 	github.com/ipfs/bbloom v0.0.4
-	github.com/ipfs/boxo v0.8.0
+	github.com/ipfs/boxo v0.8.2-0.20230602131956-922a186d9cef
 	github.com/ipfs/go-cid v0.4.0
 	github.com/ipfs/go-datastore v0.6.0
 	github.com/ipfs/go-ds-badger v0.3.0
-	github.com/ipfs/go-ipld-format v0.4.0
+	github.com/ipfs/go-ipld-format v0.4.1-0.20230530195241-c3da01c74a06
```
Comment (aschmahmann) [inline go.mod:13]: Note: need released versions

File: ipld.go [hunk 2]
```diff
@@ -15,10 +15,6 @@ import (
 
 var _ ipld.NodeGetter = (*crdtNodeGetter)(nil)
 
-func init() {
-	ipld.Register(cid.DagProtobuf, dag.DecodeProtobufBlock)
-}
```
Comment (aschmahmann) [inline ipld.go:20]: Unclear that this is used for anything in this repo. Is there anything I'm missing here?

Comment (hsanjuan) [reply ipld.go:20]: Well it seems it is not really needed. Perhaps it was needed once upon a time, or I just feared the NodeGetter wouldn't be able to parse anything. (I suspect this codec may be registered by default somewhere else?).

**Comment by hsanjuan** [comment]
LGTM, pending bubbling etc.

File: ipld.go (hunk 2 above)
Comment (aschmahmann) [reply ipld.go:20]: it used to be registered by go-merkledag, although I'm not sure it's even needed here anymore

**Pull request merged**